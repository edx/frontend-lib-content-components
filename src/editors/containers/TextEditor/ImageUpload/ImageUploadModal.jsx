import React from 'react';
import PropTypes from 'prop-types';
import { ModalDialog, ActionRow, Button } from '@edx/paragon';

const ImageUploadModal = ({ isOpen, close }) => (
  <ModalDialog
    title="My dialog"
    isOpen={isOpen}
    onClose={close}
    size="lg"
    variant="default"
    hasCloseButton
    isFullscreenOnMobile
  >
    <ModalDialog.Header>
      <ModalDialog.Title>
        Im a dialog box
      </ModalDialog.Title>
    </ModalDialog.Header>
    <ModalDialog.Body>
      <p>
        Im baby palo santo ugh celiac fashion axe.
        La croix lo-fi venmo whatever.
        Beard man braid migas single-origin coffee forage ramps.
        Tumeric messenger bag bicycle rights wayfarers, try-hard cronut blue bottle health goth.
        Sriracha tumblr cardigan, cloud bread succulents tumeric copper mug marfa semiotics woke next
        level organic roof party +1 try-hard.
      </p>
    </ModalDialog.Body>
    <ModalDialog.Footer>
      <ActionRow>
        <ModalDialog.CloseButton variant="tertiary">
          Cancel
        </ModalDialog.CloseButton>
        <Button variant="primary">
          A button
        </Button>
      </ActionRow>
    </ModalDialog.Footer>
  </ModalDialog>
);

ImageUploadModal.propTypes = {
  isOpen: PropTypes.bool,
  close: PropTypes.func,
};
ImageUploadModal.defaultProps = {
  isOpen: false,
  close: () => {},
};
export default ImageUploadModal;
