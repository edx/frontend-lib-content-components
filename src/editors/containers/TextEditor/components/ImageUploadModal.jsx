import React from 'react';
import PropTypes from 'prop-types';

import ImageSettingsModal from './ImageSettingsModal';
import SelectImageModal from './SelectImageModal';

const ImageUploadModal = ({
  // eslint-disable-next-line
  editorRef,
  isOpen,
  close,
}) => {
  // selected image file reference data object.
  // existance of this field determines which child modal is displayed
  const [selection, setSelection] = React.useState(null);
  const clearSelection = () => setSelection(null);
  const saveToEditor = (settings) => {
    // eslint-disable-next-line
    console.log({ selection, settings });
    // tell editor ref to insert content at cursor location();
  };
  const closeAndReset = () => {
    setSelection(null);
    close();
  };
  if (selection) {
    return (
      <ImageSettingsModal
        {...{
          isOpen,
          close: closeAndReset,
          selection,
          saveToEditor,
          returnToSelection: clearSelection,
        }}
      />
    );
  }
  return (<SelectImageModal {...{ isOpen, close, setSelection }} />);
};

ImageUploadModal.defaultProps = {
  editorRef: null,
};
ImageUploadModal.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  close: PropTypes.func.isRequired,
  editorRef: PropTypes.oneOfType([
    PropTypes.func,
    PropTypes.shape({ current: PropTypes.any }),
  ]),
};
export default ImageUploadModal;
