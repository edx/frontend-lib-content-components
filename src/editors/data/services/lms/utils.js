import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';

/**
 * get(url)
 * simple wrapper providing an authenticated Http client get action
 * @param {string} url - target url
 */
export const get = (...args) => getAuthenticatedHttpClient().get(...args);
/**
 * post(url, data)
 * simple wrapper providing an authenticated Http client post action
 * @param {string} url - target url
 * @param {object|string} data - post payload
 */
export const post = (...args) => getAuthenticatedHttpClient().post(...args);

export const client = getAuthenticatedHttpClient;
