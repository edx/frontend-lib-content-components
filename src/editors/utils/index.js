/* eslint-disable import/prefer-default-export */
export { keyStore, StrictDict } from '../../utils';
export { default as camelizeKeys } from './camelizeKeys';
