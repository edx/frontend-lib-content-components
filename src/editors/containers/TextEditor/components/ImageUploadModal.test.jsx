import React from 'react';
import { shallow } from 'enzyme';

import { StrictDict } from '../../../utils';

import * as module from './ImageUploadModal';
jest.mock('./ImageSettingsModal', () => 'ImageSettingsModal');
jest.mock('./SelectImageModal', () => 'SelectImageModal');

const { ImageUploadModal } = module;

describe('ImageUploadModal hooks', () => {
  describe('getImgTag', () => {
    const selection = { externalUrl: 'sOmEuRl.cOm' };
    let settings;
    let expected;
    let output;
    test('It returns a html string which matches an image tag', () => {
      settings = {
        altText: 'aLt tExt',
        isDecorative: false,
        dimensions: {
          width: 2022,
          height: 1619,
        },
      };
      expected = {
        src: selection.externalUrl,
        alt: settings.altText,
        width: settings.dimensions.width,
        height: settings.dimensions.height,
      };
      output = module.imgTag({ selection, settings });
      expect(output).toEqual(`<img ${module.propsString(expected)} />`);
    });
    test('If Is decorative is true, alt text is an empty string', () => {
      settings = {
        isDecorative: true,
        altText: 'aLt tExt',
        dimensions: {
          width: 2022,
          height: 1619,
        },
      };
      expected = {
        src: selection.externalUrl,
        alt: '',
        width: settings.dimensions.width,
        height: settings.dimensions.height,
      };
      output = module.imgTag({
        selection: selection,
        settings: settings,
      });
      expect(output).toEqual(`<img ${module.propsString(expected)} />`);
    });
  });

  describe('createSaveCallback', () => {
    const close = jest.fn();
    const execCommandMock = jest.fn();
    const editorRef = { current: { some: 'dATa', execCommand: execCommandMock } };
    const setSelection = jest.fn();
    const selection = jest.fn();
    const settings = {
      altText: 'aLt tExt',
      isDecorative: false,
      dimensions: {
        width: 2022,
        height: 1619,
      },
    };
    let output;
    beforeEach(() => {
      output = module.hooks.createSaveCallback({
        close, editorRef, setSelection, selection,
      });
    });
    afterEach(() => {
      jest.clearAllMocks();
    });
    test('It creates a callback, that when called, inserts to the editor, sets the selection to be null, and calls close', () => {
      jest.spyOn(module, 'imgTag')
        .mockImplementationOnce(({ settings }) => ({ selection, settings }));
      expect(execCommandMock).not.toBeCalled();
      expect(setSelection).not.toBeCalled();
      expect(close).not.toBeCalled();
      output(settings);
      expect(execCommandMock).toBeCalledWith('mceInsertContent', false, { selection, settings: settings });
      expect(setSelection).toBeCalledWith(null);
      expect(close).toBeCalled();
    });
  });

  describe('component', () => {
    let props;
    const hookKeys = StrictDict(Object.keys(module.hooks).reduce(
      (obj, key) => ({ ...obj, [key]: jest.fn() }),
      {},
    ));
    let hooks;
    beforeAll(() => {
      hooks = module.hooks;
      props = {
        editorRef: { current: null },
        isOpen: false,
        close: jest.fn().mockName('props.close'),
        clearSelection: jest.fn().mockName('props.clearSelection'),
        selection: { some: 'images' },
        setSelection: jest.fn().mockName('props.setSelection'),
      };
      module.hooks = {
        createSaveCallback: jest.fn().mockName('hooks.createSaveCallback'),
        onClose: jest.fn().mockName('hooks.onClose'),
      };
    });
    afterAll(() => {
      module.hooks = hooks;
    });
    test('snapshot: with selection content (ImageSettingsUpload)', () => {
      expect(shallow(<ImageUploadModal {...props} />)).toMatchSnapshot();
    });
    test('snapshot: no selection (Select Image Modal)', () => {
      expect(shallow(<ImageUploadModal {...props} selection={null} />)).toMatchSnapshot();
    });
  });
});
