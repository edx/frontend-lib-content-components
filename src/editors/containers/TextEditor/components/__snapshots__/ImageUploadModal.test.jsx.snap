// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageUploadModal component snapshot: no selection (Select Image Modal) 1`] = `
<SelectImageModal
  close={[MockFunction props.close]}
  isOpen={false}
  setSelection={[MockFunction props.setSelection]}
/>
`;

exports[`ImageUploadModal component snapshot: with selection content (ImageSettingsUpload) 1`] = `
<ImageSettingsModal
  isOpen={false}
  returnToSelection={[MockFunction props.clearSelection]}
  selection={
    Object {
      "some": "images",
    }
  }
/>
`;
